src/cgutils.cpp

```diff
@@ -988,13 +994,17 @@ static Type *julia_struct_to_llvm(jl_value_t *jt, bool *isboxed)
                 latypes.push_back(lty);
             }
             if (!isTuple) {
-                structdecl->setBody(latypes);
+                if (is_vecelement_type(jt))
+                    // VecElement type is unwrapped in LLVM
+                    jst->struct_decl = latypes[0];
+                else
+                    structdecl->setBody(latypes);
             }
             else {
                 if (isvector && lasttype != T_int1 && !type_is_ghost(lasttype)) {
                     // TODO: currently we get LLVM assertion failures for other vector sizes
-                    bool validVectorSize = (ntypes == 2 || ntypes == 4 || ntypes == 6);
-                    if (0 && lasttype->isSingleValueType() && !lasttype->isVectorTy() && validVectorSize) // currently disabled due to load/store alignment issues
+                    bool validVectorSize = (ntypes == 2 || ntypes == 4 || ntypes == 8 || ntypes == 16);
```

Comment:
There were reports that, at least for LLVM 3.3, that it did not correctly handle vectors of all lengths.  I have not tried with LLVM 3.7.  Possibly the bugs were fixed, though I suspect there's not much motivation to do so :-(.  If you want to explore relaxing the limits (and report back), you can play with this line in `src/cgutils.cpp`:
```
bool validVectorSize = (ntypes == 2 || ntypes == 4 || ntypes == 8 || ntypes == 16);
```
If we need to keep the limits, we need to think about what is the best way for Julia to detect what was actually usedl.